app/forms/idv/api_image_upload_form.rb

```diff
@@ -5,36 +5,114 @@ class ApiImageUploadForm
 
     validates_presence_of :front
     validates_presence_of :back
-    validates_presence_of :document_capture_session
     validates_presence_of :selfie, if: :liveness_checking_enabled?
+    validates_presence_of :document_capture_session
 
     validate :validate_images
     validate :throttle_if_rate_limited
 
-    def initialize(params, liveness_checking_enabled:)
+    def initialize(params, liveness_checking_enabled:, issuer:, analytics: nil)
       @params = params
       @liveness_checking_enabled = liveness_checking_enabled
+      @issuer = issuer
+      @analytics = analytics
       @readable = {}
     end
 
     def submit
       throttled_else_increment
+      @form_response = validate_form
+
+      client_response = nil
+      doc_pii_response = nil
+
+      if form_response.success?
+        client_response = post_images_to_client
+
+        if client_response.success?
+          doc_pii_response = validate_pii_from_doc(client_response)
+        else
+          client_response = client_response.merge(form_response)
+        end
+      end
+
+      return determine_response(
+        form_response: form_response,
+        client_response: client_response,
+        doc_pii_response: doc_pii_response,
+      )
+    end
+
+    private
+
+    attr_reader :params, :analytics, :issuer, :form_response
```

Comment:
I'm a bit confused by `form_response` as an instance variable. It doesn't seem like we're reading it externally, nor should we want to? Also, internally we use it in `validate_pii_from_doc`, but in `determine_response` it's passed as an argument. Could it be passed as an argument to `validate_pii_from_doc`, and remove the instance variable altogether?

Reply:
excellent point - I'll throw together another quick PR